src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -310,6 +321,41 @@
     /** Actions to add to the {@link FlowStartNode}. */
     transient final List<Action> flowStartNodeActions = new ArrayList<Action>();
 
+    enum TimingKind {
+        /**
+         * Parsing Groovy sources; includes {@link #load}.
+         * @see CpsGroovyShell#parse(GroovyCodeSource)
+         */
+        parse,
+        /**
+         * Loading classes needed during {@link #parse}.
+         * @see ClassLoader#loadClass(String, boolean)
+         * @see ClassLoader#getResource
+         */
+        load,
```

Comment:
🐛  would rename to "classLoad" because "load" could just as easily mean reading from disk/initializing (which is potentially something we might benchmark, i.e. loading program.dat read-in on resumption).